### PR TITLE
[screen-capture][ios] migrate to sweet API

### PR DESCRIPTION
**Pull request by fobos531**
# Why

A small Christmas present 🙂

# How

followed the usual migration steps

# Test Plan

Physical iPhone 14 Pro iOS 17.2

Screenshot listener + preventing screen recording **while already recording**:

https://github.com/expo/expo/assets/19533289/ac3e681f-a827-4fbb-9923-5ced2680e151

Normal recording (screen recording allowed):

https://github.com/expo/expo/assets/19533289/d3e234f0-f5ff-42ac-aa96-6a5e73c22117

3rd person view of preventing screen recording **while already recording**: 

https://github.com/expo/expo/assets/19533289/ad58f9bc-3639-4bee-816f-9582e7d531cf

Resulting video from previous case:

https://github.com/expo/expo/assets/19533289/6620537e-d152-48ef-8417-5ff4b52122c3

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [X] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-screen-capture/ios/ScreenCaptureModule.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      _blockView.frame = CGRect(x: 0, y: 0, width: boundLength, height: boundLength)
+      _blockView.backgroundColor = UIColor.black
+    }
+
+    OnStartObserving {
+      self.setIsBeingObserved(isBeingObserved: true)
+    }
+
+    OnStopObserving {
+      self.setIsBeingObserved(isBeingObserved: false)
```
Comment (alanjhughes) [inline packages/expo-screen-capture/ios/ScreenCaptureModule.swift:26]:
```suggestion
      self.setIsBeing(observed: false)
```

File: packages/expo-screen-capture/ios/ScreenCaptureModule.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    Events(onScreenshotEventName)
+
+    OnCreate {
+      let boundLength = max(UIScreen.main.bounds.size.width, UIScreen.main.bounds.size.height)
+      _blockView.frame = CGRect(x: 0, y: 0, width: boundLength, height: boundLength)
+      _blockView.backgroundColor = UIColor.black
+    }
+
+    OnStartObserving {
+      self.setIsBeingObserved(isBeingObserved: true)
```
Comment (alanjhughes) [inline packages/expo-screen-capture/ios/ScreenCaptureModule.swift:22]:
```suggestion
      self.setIsBeing(observed: true)
```

**Review by alanjhughes — CHANGES_REQUESTED** [review]
Thanks @fobos531! Just some minor changes. Also, does it prevent screenshots correctly? Your videos show blocking screen recording but I don't see any blocking a screenshot?

**Comment by fobos531** [comment]
> Thanks @fobos531! Just some minor changes. Also, does it prevent screenshots correctly? Your videos show blocking screen recording but I don't see any blocking a screenshot?

Hey @alanjhughes 

Thank you for reviewing this. I applied the changes you suggested, hopefully everything should look good now.

To answer your question about preventing screenshots, the migration preserves the exact same behavior like it has now with Objective-C. As mentioned in the Expo docs, it is not possible to entirely prevent screenshots.

I dug deeper and due to the mechanisms/event listeners Apple exposes, we can at-best make screen recordings/screenshot "unusable", but in reality we cannot prevent the screenshot/screen recording from actually being taken. Screen recordings can be easily obscured due to `UIApplication.capturedDidChangeNotification`, but since for screenshots we only have `UIApplication.userDidTakeScreenshotNotification`, it means that we're aware of the screenshot only **after** it has been taken, not **before**. Netflix, Hulu etc. can create unusable screenshots since iOS knows to obscure DRM-ed content, but for any other kind of applications various hacks need to be used. I've seen suggestions where people create a `UITextField()` and then trigger the `secureEntry` on it and stretch across the entire screen, or even more advanced, what Telegram does: https://github.com/TelegramMessenger/Telegram-iOS/blob/7c7282566f596ec9a53c14c77e5d3c82a5ae0e42/submodules/TelegramUI/Components/Stories/StoryContainerScreen/Sources/StoryItemImageView.swift#L52

Essentially, they listen for the screen capture event, and after they take the screenshot, I _assume_ that, under-the-hood, they access the captured screenshot and replace with a pre-captured snapshot of the `CaptureProtectedView` which essentially adds a layer which makes the original screenshot useless to the person that took it. All of those solutions feel hacky to some extent and I'm not sure we event want to go that route.

**Comment by alanjhughes** [comment]
Hey @fobos531 
Thanks for the clarification. That makes sense. If this is the limits of what we can do without any hacks then that is fine. Thanks again for taking the time!

**Review by alanjhughes — APPROVED** [review]

**Pull request merged**